stan/math/rev/mat/fun/determinant.hpp

```diff
@@ -27,33 +27,18 @@ class determinant_vari : public vari {
         adjARef_(
             reinterpret_cast<vari**>(ChainableStack::instance().memalloc_.alloc(
                 sizeof(vari*) * A.rows() * A.cols()))) {
-    size_t pos = 0;
-    for (size_type j = 0; j < cols_; j++) {
-      for (size_type i = 0; i < rows_; i++) {
-        A_[pos] = A(i, j).val();
-        adjARef_[pos++] = A(i, j).vi_;
-      }
-    }
+    Eigen::Map<Eigen::MatrixXd>(A_, rows_, cols_) = A.val();
+    Eigen::Map<matrix_vi>(adjARef_, rows_, cols_) = A.vi();
```

Comment:
It feels like we'll be using these maps a lot more, may be a good alias candidate